src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -84,19 +91,43 @@ void AsyncLogWriter::enqueue_locked(LogFileStreamOutput* output, const LogDecora
   _lock.notify();
 }
 
-void AsyncLogWriter::enqueue(LogFileStreamOutput& output, const LogDecorations& decorations, const char* msg) {
+// This function checks for cases where continuing with asynchronous logging may lead to stability issues, such as a deadlock.
+// If this returns true then we give up on logging asynchronously and do so synchronously instead.
+bool AsyncLogWriter::resort_to_synchronous_logging() {
```

Comment:
The name `resort_to_synchronous_logging` says something about what the callers of `enqueue` should do. I think something more in the line of `is_enqueue_disallowed` / `is_async_logging_disallowed` would be better. Or probably inverting the the logic, and have it be `is_enqueue_allowed`.